huaweicloud/services/rds/resource_huaweicloud_rds_mysql_account.go

```diff
@@ -40,182 +44,312 @@ func ResourceRdsAccount() *schema.Resource {
 				Computed: true,
 			},
 			"instance_id": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the RDS Mysql instance.`,
 			},
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the username of the DB account.`,
+			},
+			"hosts": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the IP addresses that are allowed to access your DB instance.`,
 			},
 			"password": {
-				Type:      schema.TypeString,
-				Required:  true,
-				Sensitive: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the password of the DB account.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies remarks of the DB account.`,
 			},
 		},
 	}
 }
 
-func resourceRdsAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	c := meta.(*config.Config)
-	client, err := c.HcRdsV3Client(c.GetRegion(d))
+func resourceMysqlAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createMysqlAccount: create RDS Mysql account.
+	var (
+		createMysqlAccountHttpUrl = "v3/{project_id}/instances/{instance_id}/db_user"
+		createMysqlAccountProduct = "rds"
+	)
+	createMysqlAccountClient, err := cfg.NewServiceClient(createMysqlAccountProduct, region)
 	if err != nil {
-		return fmtp.DiagErrorf("error creating RDS client: %s", err)
+		return diag.Errorf("error creating RDS client: %s", err)
 	}
 
-	dbUser := d.Get("name").(string)
 	instanceId := d.Get("instance_id").(string)
+	createMysqlAccountPath := createMysqlAccountClient.Endpoint + createMysqlAccountHttpUrl
+	createMysqlAccountPath = strings.ReplaceAll(createMysqlAccountPath, "{project_id}",
+		createMysqlAccountClient.ProjectID)
+	createMysqlAccountPath = strings.ReplaceAll(createMysqlAccountPath, "{instance_id}", instanceId)
 
-	config.MutexKV.Lock(instanceId)
-	defer config.MutexKV.Unlock(instanceId)
-
-	createOpts := &model.CreateDbUserRequest{
-		InstanceId: instanceId,
-		Body: &model.UserForCreation{
-			Name:     dbUser,
-			Password: d.Get("password").(string),
-		},
+	createMysqlAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
 	}
+	createMysqlAccountOpt.JSONBody = utils.RemoveNil(buildCreateMysqlAccountBodyParams(d))
+	log.Printf("[DEBUG] Create RDS Mysql account options: %#v", createMysqlAccountOpt)
```

Comment:
it's not advised to log the password in **Create** and **Update** options

Reply:
fixed